File: scala/src/test/scala/com/shekhargulati/ninetynine_problems/_02_arithmetic/P33Test.scala
package com.shekhargulati.ninetynine_problems._02_arithmetic

import org.scalatest.{FunSpec, Matchers}

class P33Test extends FunSpec with Matchers {

  describe("P33 Spec") {

    it("[[3,2],[5,1],[7,1]] is prime factors multiples of 315") {
      val fs = P33.primeFactorsMult(315)
      fs should be(List((3, 2), (5, 1), (7, 1)))
    }

    it("[[3,1],[11,1]] is prime factors multiples of 33") {
      val fs = P33.primeFactorsMult(33)
      fs should be(List((3, 1), (11, 1)))
    }

  }

}

File: java8/src/test/java/com/shekhargulati/ninetynine_problems/_02_arithmetic/P33Test.java
package com.shekhargulati.ninetynine_problems._02_arithmetic;

import org.junit.Test;

import java.util.AbstractMap.SimpleEntry;
import java.util.List;

import static org.hamcrest.core.IsCollectionContaining.hasItems;
import static org.junit.Assert.assertThat;

public class P33Test {

    @Test
    public void shouldFindPrimeFactorsOf315() throws Exception {
        List<SimpleEntry<Integer, Integer>> primeFactors = P33.primeFactorsMult(315);
        assertThat(primeFactors, hasItems(new SimpleEntry<>(3, 2), new SimpleEntry<>(5, 1), new SimpleEntry<>(7, 1)));
    }

    @Test
    public void shouldFindPrimeFactorsOf33() throws Exception {
        List<SimpleEntry<Integer, Integer>> primeFactors = P33.primeFactorsMult(33);
        assertThat(primeFactors, hasItems(new SimpleEntry<>(3, 1), new SimpleEntry<>(11, 1)));
    }
}